flaml/automl/task/time_series_task.py

```diff
@@ -0,0 +1,523 @@
+            )
+            if isinstance(metric_i, dict) and "intermediate_results" in metric_i:
+                del metric_i["intermediate_results"]
+            total_fold_num += 1
+            val_loss_folds.append(val_loss_i)
+            log_metric_folds.append(metric_i)
+            train_time += train_time_i
+            pred_time += pred_time_i
+            if time.time() - start_time >= budget:
+                break
+        val_loss, metric = cv_score_agg_func(val_loss_folds, log_metric_folds)
+        n = total_fold_num
+        pred_time /= n
+        return val_loss, metric, train_time, pred_time
+
+    def default_estimator_list(self, estimator_list: List[str], is_spark_dataframe: bool) -> List[str]:
+        assert not is_spark_dataframe, "Spark is not yet supported for time series"
+
+        # TODO: why not do this if/then in the calling function?
+        if "auto" != estimator_list:
+            return estimator_list
+
+        if self.is_ts_forecastpanel():
+            return ["tft"]
+
+        estimator_list = [
+            "lgbm",
+            "rf",
+            "xgboost",
+            "extra_tree",
+            "xgb_limitdepth",
+        ]
+
+        # Catboost appears to be way slower than the others, don't include it by default
+        # try:
+        #     import catboost
+        #
+        #     estimator_list.append("catboost")
+        # except ImportError:
+        #     pass
+
+        if self.is_regression():
+            estimator_list += ["arima", "sarimax"]
+
+            try:
+                import prophet
+
+                estimator_list.append("prophet")
+            except ImportError:
+                pass
+
+        return estimator_list
+
+    def default_metric(self, metric: str) -> str:
+        assert self.is_ts_forecast(), "If this is not a TS forecasting task, this code should never have been called"
+        if metric == "auto":
+            return "mape"
+        else:
+            return metric
+
+    @staticmethod
+    def prepare_sample_train_data(automlstate, sample_size):
+        # we take the tail, rather than the head, for compatibility with time series
+
+        shift = sample_size - len(automlstate.X_train.train_data)
+        sampled_X_train = automlstate.X_train.move_validation_boundary(shift)
+
+        return sampled_X_train, None, None, None
+
+
+def remove_ts_duplicates(
+    X,
+    time_col,
+):
+    """
+    Assumes the targets are included
+    @param X:
+    @param time_col:
+    @param y:
+    @return:
+    """
+
+    duplicates = X.duplicated()
+
+    if any(duplicates):
+        logger.warning("Duplicate timestamp values found in timestamp column. " f"\n{X.loc[duplicates, X][time_col]}")
+        X = X.drop_duplicates()
+        logger.warning("Removed duplicate rows based on all columns")
+        assert (
+            X[[X.columns[0]]].duplicated() is None
+        ), "Duplicate timestamp values with different values for other columns."
```

Comment:
The coverage of this part of the code gets removed by the change in the test code.